### ISSUE TITLE
Fail to build spring-music

### ISSUE THREAD

**Issue by mattysweeps** [cloudfoundry/build-system-cnb#12]
I'm trying to build [spring music](https://github.com/cloudfoundry-samples/spring-music) using [pack](https://github.com/buildpacks/pack) and the Cloud Foundry builder image, `cloudfoundry/cnb:cflinuxfs3`, particularly the digest `sha256:524783dddb5f8a3ea2b307a961a0c315ddb1d5db7d0754ad43139bf22ebee9be`.

That image contains the `org.cloudfoundry.buildsystem` buildpack at version `v1.2.9`.

When I run pack build I get the following output: https://pastebin.com/nXFU3Usk

Error:
```
[builder] Cloud Foundry Build System Buildpack v1.2.9
[builder]   unable to find built artifact (executable JAR or WAR) in build/libs/*.[jw]ar, candidates: [/workspace/build/libs/spring-music-1.0.jar /workspace/build/libs/workspace-1.0.jar]
[builder] ERROR: failed to build: exit status 103
```

Somehow two candidates are created and the buildpack does not know how to continue.
It's hard for me to troubleshoot this because I don't know how the `workspace.jar` is created. I can't recreate it locally. Any ideas on why this is happening?

**Comment by scottfrederick** [comment]
I just tried this and I don't get that error. Can you confirm that you have the very latest code from the upstream Spring Music GitHub repo, and share the `pack` command line you are using?

**Comment by mattysweeps** [comment]
Yup, ~building a gif real quick~, but I can confirm I'm up-to-date with master.
```
pack build --builder=cloudfoundry/cnb:cflinuxfs3 -v test
```

^ the exact command I run with pack `v0.8.1 (git sha: e776ebf0096363bbac60771a456af941827316be)`

**Comment by mattysweeps** [comment]
I think I figured out why the error is happening. I had already built the app locally, so there was a `./build/libs/spring-music-1.0.jar` which I'm guessing was being renamed to `workspace.jar`

Running `./gradlew clean` and then building again worked.

However, I think many users will run into this issue. Is this something Gradle is doing that the buildpack can circumvent?

**Comment by nebhale** [comment]
@mattysweeps I apparently haven't been watching this repo and missed this issue.  You're dead on that this was what the problem was (and even came up today [in Slack](https://paketobuildpacks.slack.com/archives/CULAS8ACD/p1587574796011300)).  The buildpack is oriented towards getting a clean repo (typically via Git checkout) so I'm not sure we want it to exclude this particular file explicitly.  I also can't find any explicit mention of `workspace.jar` associated with Gradle, so maybe it's something in your IDE or environment specific?
